airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source_adapter.py

```diff
@@ -41,6 +48,9 @@ def read(
         if abstract_streams:
             yield from self._concurrent_source.read(abstract_streams)
         if configured_catalog_for_regular_streams.streams:
+            if isinstance(state, MutableMapping):
```

Comment:
when would the state be a mapping instead of a list?

Reply:
The legacy state format, as I remember, could be in a mapping format

Reply:
we can deprecate support for this state type and no longer expect the old format.

Reply:
We should be able to deprecate support here. We've stopped parsing of incoming legacy state from `entrypoint.py` in this PR https://github.com/airbytehq/airbyte/pull/43459 . So for simplicity, the concurrent framework should be able to assume it won't arrive here